src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,200 @@
+                || inputStringArray[0].equals("deadline")
+                || inputStringArray[0].equals("event")) {
+                Task newTask = createNewTaskFromInput(inputStringArray[0], inputText);
+                String message = taskList.insertNewTask(newTask);
+                storage.saveData(taskList);
+                Ui.displayMessage(message);
+                return true;
+
+            } else if (inputStringArray[0].equals("delete")) {
+                try {
+                    String inputNumberString = inputStringArray[1];
+                    int deleteTaskIndex = convertToTaskIndex(inputNumberString);
+                    String message = taskList.deleteTask(deleteTaskIndex);
+                    storage.saveData(taskList);
+                    Ui.displayMessage(message);
+                    return true;
+                } catch (ArrayIndexOutOfBoundsException exception) { // no input together with command
+                    throw new DukeException(COMMAND_REQUIRES_NUMBER);
+                }
+            } else if (inputStringArray[0].equals("find")) {
+                try {
+                    String message = taskList.findTaskName(inputStringArray[1]);
+                    Ui.displayMessage(message);
+                    return true;
+                } catch (ArrayIndexOutOfBoundsException exception) {
+                    throw new DukeException(String.format(MISSING_TASK_INFO, "find"));
+                }
+            } else { // other text input
+                throw new DukeException(UNRECOGNIZED_COMMAND);
+            }
+        } catch (DukeException dukeException) {
+            Ui.displayMessage(dukeException.toString());
+            return true;
+        }
+    }
+
+    /**
+    * Returns the Task created from the user inputText.
+    * The type parameter determines which type of Task is created.
+    * Use "todo", "deadline", "event" to create the respective Task objects.
+    *
+    * @param    type          the type of Task to be created
+    * @param    inputText     input of the user
+    * @return                 the created Task
+    * @throws   DukeException throws a DukeException
+    * @see      Task
+    */
+    public static Task createNewTaskFromInput(String type, String inputText) throws DukeException {
+        Task newTask;
+        String taskName;
+        String taskDateTime;
+        String missingDateTime = "Sorry Sir, the description of <" + type + "> is missing a date/time.";
+        String missingTaskInfo = "Sorry Sir, the <" + type + "> command cannot be empty.";
+        String wrongDateTimeFormat = "Sorry Sir, the date/time needs to be in the format: YYYY-MM-DD HH:MM";
+
+        if (type.equals("deadline")) { // is a deadline task
+            String[] inputStringArray = inputText.split(" /by ");
+            try {
+                taskName = inputStringArray[0].substring(9);
+            } catch (StringIndexOutOfBoundsException exception) {
```

Comment:
An alternative solution to this separating out the commands into their own individual classes that inherit from a Command Parent class, and in the classes subsequently parse the input to get the output required. E.g DeadlineCommand class and a runCommand() method to splice the string and return the required output. Would be geared toward a more object oriented design. Otherwise, this block of code was easy to understand too! :)